File: src/2022/day01/src/main/java/ca/terrylockett/aoc2022/day01/Day01Runner.java
package ca.terrylockett.aoc2022.day01;

import ca.terrylockett.aoccommon.inputfilefinder.Resources;

import java.io.FileNotFoundException;
import java.net.URISyntaxException;

public class Day01Runner {

	public static void main(String[] args) throws URISyntaxException, FileNotFoundException {

		String filePath = Resources.getInputFilePath("input.txt").orElseThrow();

		// ***** part 01 *****//
		int maxCals = DumbElfCalorieTracker.findTheFoodHog(filePath);
		System.out.println("part1 - biggest calorie total: " + maxCals);

		// ***** part 02 *****//
		int threeMaxCals = DumbElfCalorieTracker.findTheFoodHog(filePath, 3);
		System.out.println("part2 - Three biggest calorie totals: " + threeMaxCals);
	}
}

File: src/2023/day01/src/main/kotlin/ca/terrylockett/aoc2023/day01/Day01Runner.kt
package ca.terrylockett.aoc2023.day01

import ca.terrylockett.aoccommon.inputfilefinder.Resources

fun main() {
	val inputFile: String = Resources.getInputFilePath("input.txt").orElseThrow()

	println("2023 day01 part1: ${CalibrationDocument.basicCalibration(inputFile)}")
	println("2023 day01 part2: ${CalibrationDocument.fancyCalibration(inputFile)}")
}
